### DISCUSSION TITLE
Allow the user to store images to be used in the screens

### DISCUSSION THREAD

**Discussion by willi-piske** [Budibase/budibase#5278] [Ideas]
Hi,

I think in cases like when you want to show a custom image, icon or illustration it would be much more straightforward and performatic to upload the image file to the budibase's minio instance or to the database as an attachment and be able to load it into an image component rather than having to upload it to an external service and provide the URL to the image component.

I've been employing a workaround to do that which consists in:

1. saving the image file in the internal database as an attachment;
2. placing an image component inside a data provider and;
3. using handlebars to provide the attachment url to the image component.

If there's a simpler way to accomplish that, please let me know, but in anyway I can think of a number of ways to make it much more user friendly, like having an _assets_ tab in the budibase admin page where one can save files and then allowing the user to provide an URL or browsing and picking an asset from the image component settings tab.

Thanks a lot!

**Comment by melohagan** [comment]
Hi @willi-piske 

Great idea! I've raised a feature request here: https://github.com/Budibase/budibase/issues/5285
